File: native_library/apple/FirebaseANE/FirestoreANE/Extensions/FreDocumentChange.swift
import Foundation
import FirebaseFirestore

public extension DocumentChange {
    func toDictionary() -> [String: Any] {
        var ret = [String: Any]()
        ret["type"] = self.type.rawValue
        ret["newIndex"] = self.newIndex
        ret["oldIndex"] = self.oldIndex
        ret["id"] = self.document.documentID
        return ret
    }
}

File: native_library/android/FirebaseANE/Firestore/src/main/java/com/tuarua/firebase/firestore/extensions/FreDocumentChange.kt
package com.tuarua.firebase.firestore.extensions

import com.google.firebase.firestore.DocumentChange

fun DocumentChange.toMap(): Map<String, Any?>? {
    return mapOf(
            "type" to this.type.ordinal,
            "newIndex" to this.newIndex,
            "oldIndex" to this.oldIndex,
            "documentId" to this.document.id)
}
